src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -22,15 +22,29 @@ export const portfolio = createSlice({
       moduleLogger.info('clearing portfolio')
       return initialState
     },
+    setWalletId: (state, { payload }: { payload: WalletId | undefined }) => {
+      // note this function can unset the walletId to undefined
+      moduleLogger.info(payload, 'setting wallet id')
+      state.walletId = payload
+      if (!payload) return
+      state.wallet.ids = Array.from(new Set([...state.wallet.ids, payload]))
+    },
     upsertAccountMetadata: (state, { payload }: { payload: AccountMetadataById }) => {
       moduleLogger.debug('upserting account metadata')
       state.accountMetadata.byId = {
         ...state.accountMetadata.byId,
         ...payload,
       }
-      state.accountMetadata.ids = Array.from(
-        new Set([...state.accountMetadata.ids, ...Object.keys(payload)]),
-      )
+      const accountIds = Object.keys(payload)
+      state.accountMetadata.ids = Array.from(new Set([...state.accountMetadata.ids, ...accountIds]))
+
+      if (!state.walletId) return
+      const existingWalletAccountIds = state.wallet.byId[state.walletId] ?? []
+      // keep an index of what account ids belong to this wallet
+      state.wallet.byId = {
+        ...state.wallet.byId,
+        [state.walletId]: Array.from(new Set([...existingWalletAccountIds, ...accountIds])),
```

Comment:
and couldn't this just be set to `state.accountMetata.ids` after it is updated? or the same value from `Object.keys(state.accountMetadata.byId)`?